sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -4519,11 +4519,13 @@ public ConsistencyLevel getDefaultConsistencyLevelOfAccount() {
     @Override
     public void configureFaultInjectorProvider(IFaultInjectorProvider injectorProvider) {
         checkNotNull(injectorProvider, "Argument 'injectorProvider' can not be null");
+
         if (this.connectionPolicy.getConnectionMode() == ConnectionMode.DIRECT) {
-            this.storeModel.configureFaultInjectorProvider(injectorProvider);
-        } else {
-            throw new IllegalArgumentException("configureFaultInjectorProvider is not supported for gateway mode");
+            this.storeModel.configureFaultInjectorProvider(injectorProvider, this.configs);
+            this.addressResolver.configureFaultInjectorProvider(injectorProvider, this.configs);
         }
+
+        this.gatewayProxy.configureFaultInjectorProvider(injectorProvider, this.configs);
```

Comment:
This will always be configured now, even if the fault injector is in direct mode, right?

Reply:
yes, because no matter whether gateway or direct, there are always requests going to gateway